src/components/SignIn/index.jsx

```diff
@@ -0,0 +1,162 @@
+    firebase
+      .auth()
+      .signInWithPopup(googleProvider)
+      .then(function (result) {
+        setAlert({
+          message: "signIn.successMsg",
+          status: "success",
+        });
+        history.push("/");
+      })
+      .catch(function (error) {
+        setAlert({
+          message: error.message,
+          status: "danger",
+        });
+      });
+  };
+
+  const handleInput = (e) => {
+    const value = e.target.value;
+    if (e.target.type === "email") {
+      setEmail(value);
+    } else if (e.target.type === "password") {
+      setPassword(value);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    firebase
+      .auth()
+      .signInWithEmailAndPassword(email, password)
+      .then((user) => {
+        setAlert({
+          message: "signIn.successMsg",
+          status: "success",
+        });
+        history.push("/");
+      })
+      .catch(function (error) {
+        setAlert({
+          message: error.message,
+          status: "danger",
+        });
+      });
+  };
+  const { t } = useTranslation();
+
+  return (
+    <Container className="container justify-content-center">
+      <Row className="justify-content-center">
+        <h4 className="welcomeBack">{t("signIn.welcomeBack")}</h4>
+      </Row>
+      <Row className="justify-content-center">
+        <p>{t("signIn.signInText")}</p>
+      </Row>
+      <Row className="justify-content-center">
+        <div className="icons">
+          <Button
+            className="socialIcons"
+            onClick={handleGoogleLogin}
+            variant="danger"
+          >
+            <FontAwesomeIcon icon={faGoogle} />
+          </Button>
+          <Button className="socialIcons" onClick={handleFacebookLogin}>
+            <FontAwesomeIcon icon={faFacebook} />
+          </Button>
+        </div>
+      </Row>
+      <Form id="signInForm" onSubmit={handleSubmit}>
+        <Form.Group as={Row} className="justify-content-center">
+          <p>{t("signIn.or")}</p>
+        </Form.Group>
+        <Form.Group
+          as={Row}
+          className="justify-content-center"
+          controlId="formBasicEmail"
+        >
+          <Col xs={5}>
+            <Form.Control
+              onChange={handleInput}
+              type="email"
+              placeholder={t("signIn.enterEmail")}
+            />
+            <Form.Text className="text-muted">{t("signIn.emailMsg")}</Form.Text>
+          </Col>
+        </Form.Group>
+
+        <Form.Group
+          as={Row}
+          className="justify-content-center"
+          controlId="formBasicPassword"
+        >
+          <Col xs={5}>
+            <Form.Control
+              onChange={handleInput}
+              type="password"
+              placeholder={t("signIn.password")}
+            />
+          </Col>
+        </Form.Group>
+        <Row className="justify-content-center">
+          {alert && <Alert variant={alert.status}>{t(alert.message)}</Alert>}
```

Comment:
for the t(alert.message) I don't see it having the case when it gives u an error message, please check line 88